protos/sources/vega/data_source.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
```

Comment:
Why move all these in here instead of keeping it in the vega/data package?

Reply:
Because when the top level object - DataSourceDefinition is used across the code now - it is handled by interfaces to interpret the predefined data types - like it is with proposal changes - that can be different types of structs, and Product /Future.
But these types - DataSourceDefinition, DataSourceDefinionExternal/Internal - need to be also handled in the datanode side -> https://github.com/vegaprotocol/vega/blob/vega-fix-dsrc/datanode/gateway/graphql/schema.graphql#L1448

In order the graphql interfaces to satisfy the new ones from the core side - with the unions (as it is with ProposalTerms for example) they need to be read from here -> https://github.com/vegaprotocol/vega/blob/vega-fix-dsrc/protos/vega/vega.go.
And for that they need to be defined within the `vega` package, but not the `vega.data` one.
I moved the top level ones definitions outside the data, as well as DataSourceSpec because it would create a cycle import if it is defined in the vega.data package.